client/src/components/Atoms/Confirm.test.tsx

```diff
@@ -0,0 +1,94 @@
+import React from 'react'
+import { render, screen, within, waitFor } from '@testing-library/react'
+import userEvent from '@testing-library/user-event'
+import { Intent } from '@blueprintjs/core'
+import { useConfirm, Confirm, IConfirmOptions } from './Confirm'
+
+const confirmMock: IConfirmOptions = {
+  title: 'Confirm Component',
+  description: (
+    <div>
+      <p>
+        <strong>This action cannot be undone.</strong>
+      </p>
+    </div>
+  ),
+  yesButtonLabel: 'Yes',
+  yesButtonIntent: Intent.PRIMARY,
+  onYesClick: () =>
+    new Promise<void>(resolve => {
+      resolve()
+    }),
+}
+
+jest.mock('react', () => ({
+  ...jest.requireActual('react'),
+  useState: jest.fn(),
+}))
+
+afterEach(() => {
+  jest.clearAllMocks()
+})
+
+describe('Confirm component', () => {
```

Comment:
I think it might be simpler to test by creating a "consumer" component that mimics the way the `Confirm` component and `useConfirm` hooks interact in their real usage. Here's an example of what that could look like:

```ts
describe('Confirm', () => {
  it('shows a confirmation dialog', async () => {
    const onYesClickMock = jest.fn().mockResolvedValue(true)

    const ConfirmConsumer = () => {
      const { confirm, confirmProps } = useConfirm()
      return (
        <>
          <Button
            onClick={() =>
              confirm({
                title: 'Test Title',
                description: 'Test description',
                onYesClick: onYesClickMock,
              })
            }
          >
            Open Confirm
          </Button>
          <Confirm {...confirmProps} />
        </>
      )
    }
    render(<ConfirmConsumer />)

    userEvent.click(screen.getByRole('button', { name: 'Open Confirm' }))
    const dialog = screen
      .getByRole('heading', { name: 'Test Title' })
      .closest('.bp3-dialog') as HTMLElement
    userEvent.click(within(dialog).getByRole('button', { name: 'Ok' }))
    expect(onYesClickMock).toHaveBeenCalled()
    await waitFor(() => expect(dialog).not.toBeInTheDocument())
    // ... more interactions and assertions ...
  })
})
```